openapi/openapi.yml

```diff
@@ -609,6 +609,12 @@ paths:
                   items:
                     type: string
               additionalProperties: false
+              required:
```

Comment:
While we were writing endpoints for Azure, resources had both PUT and PATCH endpoints. The PUT update required all parameters, while PATCH only updated the provided parameters. I'm not sure if we should require all parameters. Do we allow partial updates? cc @byucesoy @furkansahin

Reply:
Currently, all of the properties listed are required in the route, so we should definitely require them in openapi.yml as well. It is possible to make them all optional with changes to the route, but I'm not sure whether that is worth doing.